devicons/devicon
# Feature: Peek and Build-Bot Upgrade

Hello guys,

This is a big upgrade of our build/peek workflow. I thought of splitting it up into multiple PRs but there are a lot of similar aspects (`peek-bot` and `build-bot` has the same superclass, the `footer` upgrade is a part of the `build-bot`). I apologize in advance if this is a lot and feel free to take your time to review this. 

Here is a list of features according to categories:

`peek-bot`:
1. Feature:
	- Now let us know whether an icon contains strokes when uploading them to Icomoon. This is a secondary check to an upcoming checkSvg update
	- Take snapshots of the ICONS version generated by Icomoon. The images seen on the "Generate Font" page of Icomoon is actually what the final font version look like. `peek-bot` now takes a picture of that.
	- Improved messaging to display the images
	- PR title can now be either "new icon" or "update icon". This is merely a semantic change so it looks clearer in the PR tab and release message
2. Things maintainer should know:
	- The glyph counts should match the number of icons selected (see the `peek-bot`'s PR message, I added a list of things we should look out for). If not, something went wrong => could be because color is not stripped properly.
3. Test:
	- See [this PR](https://github.com/Thomas-Boi/devicon/pull/44#issuecomment-894983099) to see a test case where there are NO STROKES in the SVG.
	- See [this PR](https://github.com/Thomas-Boi/devicon/pull/50#issuecomment-894983025) to see a test case where there are STROKES in the SVG.
	
	
`build-bot`:
1. Feature:
	- Now takes picture of the icons generated by Icomoon
	- Also checks for strokes in SVGs and duplicated icons. If these happen, build script will fail.
	- Incorporate `get_release_message` workflow into the `build-bot`. We don't have to run the two scripts separately anymore.
	- `optimizer-bot` (a part of `build-bot`):
		- Will append a `\n` to each SVG after it's optimized (looking at you *nix users)
		- Will prefix the classes -> SVGs can now use classes and `<style>` tags with no issues.
2. Things maintainer should know:
	- The build release message is now included in the PR message. You can copy it by hovering over the top right corner of the release message section.
	- ![image](https://user-images.githubusercontent.com/43018778/129284550-2f8a79d1-bdce-4819-9963-a8290f2a71a7.png)

	- Please check the files via "Edit Files" to see the `\n` in the SVG files. For some reason, GitHub don't display them but they are there. You can download the file and check locally as well => the `\n` will be there.
	- ![image](https://user-images.githubusercontent.com/43018778/129284572-068c9a88-00aa-4366-a3bb-a94f2868236d.png)
	- ![image](https://user-images.githubusercontent.com/43018778/129284602-8f6a4110-81aa-4937-83dd-fc6f9750bc4f.png)

3. Test:
	- See [this PR](https://github.com/Thomas-Boi/devicon/pull/69) to see the `built-bot`'s result.
	- [Here's](https://github.com/Thomas-Boi/devicon/actions/runs/1125673312) the Actions log for it.

Both the upgrades require merge into `master` due to the `workflow_run` events in the related workflow files.

After this PR, I have another one for `checkSvg`. After that, we can create a new release.
